### PR TITLE
route_stop_patterns

### PR DESCRIPTION
**Pull request by gknisely**
Added shape to transit.

note in transitbuilder - distances are saved to a vector.  A map stores the points and begin and end indexes and is looked up via a shape_id.

OLD:
image

NEW:
image

### CODE REVIEW

File: src/mjolnir/transit_fetcher.cc [hunk 1]
```diff
@@ -506,6 +553,20 @@ void fetch_tiles(const ptree& pt, std::priority_queue<weighted_tile_t>& queue, u
       request = response.get_optional<std::string>("meta.next");
     } while(request && (request = *request + api_key));
 
+    std::unordered_map<std::string, size_t> shapes;
```
Comment (kevinkreiser) [inline src/mjolnir/transit_fetcher.cc]: for the sake of symmetry itd be cool if you could put a comment above this block discribing what its fetching from the api

**Comment by gknisely** [comment]
updated

File: src/mjolnir/transit_fetcher.cc [hunk 2]
```diff
@@ -272,6 +273,29 @@ void get_routes(Transit& tile, std::unordered_map<std::string, uint64_t>& routes
...
+    auto* shape = tile.add_shapes();
+    auto shape_id = shape_pt.second.get<std::string>("onestop_id");
+
+    std::vector<PointLL> trip_shape;
+    PointLL  ll;
+    for(const auto& geom : shape_pt.second.get_child("geometry.coordinates")) {
+      auto lon = geom.second.front().second.get_value<float>();
+      auto lat = geom.second.back().second.get_value<float>();
+      ll.Set(lon,lat);
+      trip_shape.push_back(ll);
```
Comment (kevinkreiser) [inline src/mjolnir/transit_fetcher.cc]: emplace_back (lon,lat) saves some copies

**Comment by kevinkreiser** [comment]
Dank je wel

**Comment by dnesbitt61** [comment]
:shipit:

**Pull request merged**